src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/replication.py

```diff
@@ -0,0 +1,108 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from .azure.mgmt.containerregistry.models import Replication
+
+from ._factory import get_acr_service_client
+from ._utils import get_resource_group_name_by_registry_name
+
+import azure.cli.core.azlogging as azlogging
+logger = azlogging.get_az_logger(__name__)
+
+def acr_replication_list(registry_name,
+                         resource_group_name=None):
+    '''Lists all the replications for the specified container registry.
+    :param str registry_name: The name of container registry
+    '''
+    if resource_group_name is None:
+        resource_group_name = get_resource_group_name_by_registry_name(registry_name)
```

Comment:
Can you "functionize" - resource_group_name  = get_resource_group_name(registry, resource_group_name)   and do the If check only in that function?